Document/PostManager.php

```diff
@@ -14,28 +14,63 @@
 use Sonata\CoreBundle\Model\BaseDocumentManager;
 use Sonata\DoctrineMongoDBAdminBundle\Datagrid\Pager;
 use Sonata\DoctrineMongoDBAdminBundle\Datagrid\ProxyQuery;
+use Sonata\NewsBundle\Model\BlogInterface;
 use Sonata\NewsBundle\Model\PostManagerInterface;
 
 class PostManager extends BaseDocumentManager implements PostManagerInterface
 {
     /**
-     * @param $year
-     * @param $month
-     * @param $day
-     * @param $slug
-     *
-     * @return mixed
+     * {@inheritdoc}
      */
-    public function findOneBySlug($year, $month, $day, $slug)
+    public function findOneByPermalink($permalink, BlogInterface $blog)
     {
-        $pdqp = $this->getPublicationDateQueryParts(sprintf('%s-%s-%s', $year, $month, $day), 'day');
-
-        return $this->getRepository()
-            ->createQueryBuilder()
-            ->field('slug')->equals($slug)
-            ->andWhere($pdqp['query'])
-            ->getQuery()
-            ->getSingleResult();
+        $repository = $this->getRepository();
+
+        $query = $repository->createQueryBuilder('p');
+
+        try {
+            $urlParameters = $blog->getPermalinkGenerator()->getParameters($permalink);
+
+            $parameters = array();
+
+            if (isset($urlParameters['year']) && isset($urlParameters['month']) && isset($urlParameters['day'])) {
+                $pdqp = $this->getPublicationDateQueryParts(sprintf('%d-%d-%d', $urlParameters['year'], $urlParameters['month'], $urlParameters['day']), 'day');
+
+                $parameters = array_merge($parameters, $pdqp['params']);
```

Comment:
`$parameters` was just initialized to `[]`, so `$parameters = array($queryParts['params']);` should be enough. If you disagree because you think people might add parameters before in the future, just write `$parameters +=$queryParts['params'];`, no need for `array_merge` here.